bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DoubleByteBuf.java

```diff
@@ -370,7 +371,10 @@ public ByteBuffer nioBuffer(int index, int length) {
 
     @Override
     public ByteBuffer[] nioBuffers(int index, int length) {
-        return new ByteBuffer[] { nioBuffer(index, length) };
+        if (b1.nioBufferCount() == 1 && b2.nioBufferCount() == 1) {
+            return new ByteBuffer[] { b1.nioBuffer(), b2.nioBuffer() };
+        }
+        return ObjectArrays.concat(b1.nioBuffers(), b2.nioBuffers(), ByteBuffer.class);
```

Comment:
with this change we are no more observing index and length parameters
@merlimat  @ivankelly  @sschepens

Reply:
yup, this is incorrect, seems I failed somewhere, this logic has to be in `public ByteBuffer[] nioBuffers()`, gonna change it

Reply:
Just checked and it's OK on the Pulsar, seems you modified the incorrect method :smile:

Reply:
Oh thank you @sschepens  I have messed up with the diff, thank you very much